src/display/ui_state.rs

```diff
@@ -51,38 +51,61 @@ pub struct UIState {
 }
 
 impl UIState {
+    fn get_proc_name<'a>(
+        connections_to_procs: &'a HashMap<LocalSocket, String>,
+        local_socket: &LocalSocket,
+    ) -> Option<&'a String> {
+        if let Some(process_name) = connections_to_procs.get(local_socket) {
+            Some(process_name)
+        } else if let Some(process_name) = connections_to_procs.get(&LocalSocket {
+            ip: IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
```

Comment:
Technically, we don't need the extra enum. Because IpAddr itself knows whether it is v4 or v6.  We can figure out whether a local socket is v4 or v6, but still, we must conditionally create a v4 wildcard(0.0.0.0) or a v6 wildcard(::0).